frontend/src/components/common/RouterTabs.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useMemo } from "react";
+import {
+  generatePath,
+  matchPath,
+  Route,
+  type RouteProps,
+  Switch,
+  useHistory,
+  useLocation,
+  useParams,
+} from "react-router-dom";
+import { Tab, TabList, TabPanel, TabPanels, Tabs } from "@chakra-ui/react";
+
+type RouteConfig = {
+  name?: string;
+  path: string;
+  component: React.ComponentType;
+} & Omit<RouteProps, "component" | "render" | "children">;
+
+type RouterTabsProps = {
+  routes: RouteConfig[];
+};
+
+const RouterTabs = ({ routes }: RouterTabsProps) => {
+  const history = useHistory();
+  const matchParams = useParams();
+  const location = useLocation();
+  const currentRouteIndex = useMemo(
+    () => routes.findIndex((route) => matchPath(location.pathname, route)),
+    [routes, location.pathname],
+  );
+
+  return (
+    <>
+      <Tabs
+        index={currentRouteIndex}
+        onChange={(index) =>
+          history.push(generatePath(routes[index].path, matchParams))
+        }
+      >
+        <TabList>
+          {routes.map(({ name, path }) =>
+            name ? <Tab key={path}>{name}</Tab> : null,
+          )}
+        </TabList>
+        <TabPanels>
+          {routes.map(({ path, component: Component }) => (
+            <TabPanel key={path} p={0} pt={8}>
+              <Component />
+            </TabPanel>
+          ))}
+        </TabPanels>
+        <Switch>
+          {routes.map(({ path }) => (
+            <Route key={path} path={path} />
+          ))}
+        </Switch>
+      </Tabs>
+    </>
+  );
+};
+
+export default RouterTabs;
```

Comment:
nice.

i wonder if we should have a general tabs component? & then a component that uses that to handle when tabs have separate routes. cuz our existing tab stuff is pretty messy

although, for a separate PR maybe, but would love to hear your thoughts of whether you think we should create a ticket for this 👀

Reply:
I'm not sure how useful a separate non-router tabs component would be, since the built-in chakra tabs components are already pretty clean? unless you mean that you'd like to have a data-integrated component for some of those multi-tab data tables? in which case yeah definitely a different ticket since this pr doesn't even use a data table